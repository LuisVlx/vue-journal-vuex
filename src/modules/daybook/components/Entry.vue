<template>
    <div
        class="entry-container mb-3 pointer p-2"
        @click="$router.push({name: 'entry', params: {id: 10}})"
    >
        <!-- TÃ­tulo -->
        <div class="entry-title d-flex">
            <span class="text-success fs-5 fw-bold">24</span>
            <span class="mx-1 fs-5">Febrero</span>
            <span class="mx-2 fw-light">2023, viernes</span>
        </div>

        <div class="entry-description">
            Id elit proident voluptate qui minim anim. 
            Tempor irure ea commodo proident nulla. 
            Nisi culpa veniam commodo mollit deserunt 
            occaecat pariatur magna nostrud pariatur 
            nisi elit. Ullamco nulla nulla nostrud in. 
            Incididunt nostrud eu anim esse.
        </div>
    </div>
</template>

<script>
export default {

}
</script>

<style lang="scss" scoped>
.entry-container{
    border-bottom: 1px solid #2c3e50;
    transition: 0.2s all ease-in;

    &:hover{
        background-color: lighten($color: grey, $amount: 45);
        transition: 0.2s all ease-in;
    }

    .entry-description{
        font-size: 12px;
    }
}
</style>