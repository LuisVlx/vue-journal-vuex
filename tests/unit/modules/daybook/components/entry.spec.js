import { shallowMount } from "@vue/test-utils";

import Entry from '@/modules/daybook/components/Entry.vue'
import { journalState } from "../../../mock-data/test-journal-state";

describe('Entry Component', () => {
    
    const mockRouter = {
        push: jest.fn()
    }

    const wrapper = shallowMount(Entry, {
        props:{
            entry: journalState.entries[0]
        },
        global:{
            mocks: {
                $router: mockRouter
            }
        }
    })

    test('debe de hacer match con el snapshot ', () => {
        
        expect(wrapper.html).toMatchSnapshot()

    });

    test('debe de redireccionar al hacer click en el entry-container ', () => {
        
        const entryContainer = wrapper.find('.entry-container')

        entryContainer.trigger('click')

        expect(mockRouter.push).toHaveBeenCalledWith(
            {
                name: 'entry', 
                params: {
                    id: journalState.entries[0].id
                }
            }
        )

    });

    test('pruebas en las propiedades computadas', () => {
        
        expect(wrapper.vm.day).toBe(1)
        expect(wrapper.vm.month).toBe('Marzo')
        expect(wrapper.vm.yearDay).toBe('2023, Mi√©rcoles')

    });

});

