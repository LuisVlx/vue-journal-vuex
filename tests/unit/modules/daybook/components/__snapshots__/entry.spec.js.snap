// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entry Component debe de hacer match con el snapshot  1`] = `[Function]`;
